flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1130,6 +1131,7 @@ private void connectToResourceManager() {
 			getAddress(),
 			getResourceID(),
 			unresolvedTaskManagerLocation.getDataPort(),
+			JMXService.getPort().orElse(0),
```

Comment:
This strikes me as an odd default value, and I could imagine this being misunderstood as "running on some random port". `-1` may be a better choice, or even better, with my above suggestion of using a map, just not including it in the map at all.